### ISSUE TITLE
New Telemetry Level should respect my prior settings

### ISSUE THREAD

**Issue by john-aws** [microsoft/vscode#134660]
As of VS Code 1.61, there is a new `telemetry.telemetryLevel` setting. This defaults to 'on'. It would be preferable for the default to respect my prior settings - if all of the prior telemetry settings were disabled, then I clearly prefer no telemetry and VS Code would ideally respect that when migrating from the old settings.

**Comment by lramos15** [comment]
@john-aws It does respect your prior settings. If you disabled telemetry before it will still be disabled. You can confirm this by setting Log level to trace and seeing no telemetry is flowing in the output channel. The old setting will never be completely removed to prevent enablement of people who previously disabled telemetry but it is deprecated so we recommend setting this new setting to "off" and removing the older settings from your `settings.json`

**Comment by sander1095** [comment]
Then why the is the text still "on". It causes a lot of confusion online

**Comment by Tyriar** [comment]
FYI thid came up on HN as well today, let's make sure we make this messaging clear https://news.ycombinator.com/item?id=28812486

**Comment by judge2020** [comment]
For reference, this is the code:

https://github.com/microsoft/vscode/blob/be75065e817ebd7b6250a100cf5af78bb931265b/src/vs/platform/telemetry/common/telemetryUtils.ts#L120-L137

If the old config exists and is 'off', it sends no telemetry. Otherwise, it uses the new setting's value. 


As feedback, the setting UI for telemetry should do its own check to see if `oldConfig === undefined` and tell the user that their current setting is being respected. If it gets deprecated and removed, perhaps there should be a migration to set the new value to the old value.

**Comment by judge2020** [comment]
@Ranguna I don’t think you could make the case for telemetry being PII, so GDPR likely wouldn’t apply to this data; maybe it could be considered under ePrivacy but that's a different set of laws.

**Comment by lramos15** [comment]
I have updated the code to migrate the old settings to the new ones. While this doesn't cause any behavior change as @judge2020 pointed out, this will prevent the UI confusion that caused this thread to occur in the first time. We had intentions to add this code in the coming versions but wanted to transition slowly, clearly that was a mistake and caused more confusion than necessary. Feel free to look at the commit (you'll even see some commented out code with our plan to add this later) and let me know if you have any more issues. I'm leaving this issue open as we are also aiming to clarify the documentation some more as the team aims to be transparent. This feature was introduced to give you more control over your telemetry not less and I apologize for all the confusion this caused.
